public/app/features/dashboard-scene/settings/AnnotationsEditView.tsx

```diff
@@ -51,11 +51,32 @@ export class AnnotationsEditView extends SceneObjectBase<AnnotationsEditViewStat
     return this._dashboard;
   }
 
-  public onNew = () => {
+  // Get the datasource that supports annotations
+  public getDataSourceRefForAnnotation = async () => {
+    // get current default datasource ref from instance settings
+    const defaultInstanceDS = getDataSourceSrv().getInstanceSettings(null);
+    // get datasource ref from default datasource
+    const defaultDSRef = getDataSourceRef(defaultInstanceDS!);
```

Comment:
I would not assume there is always a default instance here. Either the TS is wrong, or there is a possibility in code to get a falsy value here. So, following the implementation, I'd control what happens in case this is a falsy value. WDYT?